addons/mail/views/mail_activity_views.xml

```diff
@@ -182,4 +182,103 @@
+                                <div>
+                                    <div t-attf-class="#{record.active.raw_value ? '' : 'o_text_line_through'} o_reminder_title">
+                                        <strong><field name="summary"/></strong>
+                                    </div>
+                                    <div class="row">
+                                        <div class="col-xs-10">
+                                            <span class="text-muted"><span class="fa fa-clock-o"/> <field name="date_deadline"/></span>
+                                        </div>
+                                        <div class="col-xs-2">
+                                            <a name="toggle_active" type="object" t-if="record.active.raw_value" title="Mark as done" class="pull-right"><i class="fa fa-check"/></a>
+                                            <a name="toggle_active" type="object" t-if="!record.active.raw_value" title="Re-open reminder" class="pull-right"><i class="fa fa-undo"/></a>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </t>
+                </templates>
+            </kanban>
+        </field>
+    </record>
+
+    <record id="mail_activity_view_search_reminder" model="ir.ui.view">
+        <field name="name">mail.activity.reminder.view.search</field>
+        <field name="model">mail.activity</field>
+        <field name="arch" type="xml">
+            <search string="Activity">
+                <field name="active"/>
+                <field name="note"/>
+                <separator/>
+                <filter string="Active" name="filter_open_reminders"
+                        domain="[('active', '=', True)]"/>
+                <filter string="Archived" name="filter_closed_reminders"
+                        domain="[('active', '=', False)]"/>
+                <separator/>
+                <filter string="Late Reminders" name="activities_overdue" domain="[('date_deadline', '&lt;', context_today().strftime('%Y-%m-%d'))]" help="Show all reminders for which the next action date is before today"/>
+                <filter string="Today Reminders" name="activities_today" domain="[('date_deadline', '=', context_today().strftime('%Y-%m-%d'))]"/>
+                <filter string="Future Reminders" name="activities_upcoming_all" domain="[('date_deadline', '&gt;', context_today().strftime('%Y-%m-%d'))]"/>
+            </search>
+        </field>
+    </record>
+
+    <record id="mail_activity_action_reminders" model="ir.actions.act_window">
+        <field name="name">Reminders</field>
+        <field name="res_model">mail.activity</field>
+        <field name="view_type">form</field>
+        <field name="search_view_id" ref="mail_activity_view_search_reminder"/>
+        <field name="view_ids"
```

Comment:
I am now more in favor of defining the ir.actions.act_window.view subelements. It eases the customization if for example you want to change the view used for kanban using standard inheritance. It is a bit verbose but it gives more flexibility.